src/test/java/marquez/db/DatasetDaoTest.java

```diff
@@ -77,13 +100,36 @@ public void testInsert() {
     assertThat(rowsAfter).isEqualTo(rowsBefore + 1);
   }
 
+  @Test(expected = UnableToExecuteStatementException.class)
+  public void testIdempotentInsert() {
+    final int rowsBefore = datasetDao.count();
+
+    final DatasetRow newDatasetRow =
+        newDatasetRowWith(namespaceRow.getUuid(), datasourceRow.getUuid());
+    datasetDao.insert(newDatasetRow);
+
+    final int rowsAfter = datasetDao.count();
+    assertThat(rowsAfter).isEqualTo(rowsBefore + 1);
+
+    datasetDao.insert(newDatasetRow);
+  }
+
+  @Test(expected = UnableToExecuteStatementException.class)
+  public void testIdempotentInsertAndGet() {
+    final int rowsBefore = datasetDao.count();
+
+    final DatasetRow newDatasetRow =
+        newDatasetRowWith(namespaceRow.getUuid(), datasourceRow.getUuid());
+    datasetDao.insert(newDatasetRow);
+
+    final int rowsAfter = datasetDao.count();
+    assertThat(rowsAfter).isEqualTo(rowsBefore + 1);
+
+    datasetDao.insertAndGet(newDatasetRow);
```

Comment:
Are you expecting `datasetDao.insertAndGet()` to throw an exception?

Reply:
Yup